runtime/shared_common/OSCachemmap.cpp

```diff
@@ -988,6 +991,13 @@ SH_OSCachemmap::internalAttach(bool isNewCache, UDATA generation)
 	}
 #endif
 
+#if defined(J9ZOS39064)
+	if (J9_ARE_NO_BITS_SET(_runtimeFlags, J9SHR_RUNTIMEFLAG_MAP31)
+		&& zos_version_at_least(ZOS_V2R4_RELEASE, ZOS_V2R4_VERSION)
```

Comment:
Hang, you are saying zos_version_at_least() is a compile time check?

Also I'm thinking we should fix OMR to allow __MAP_64 when running on V2.4 or later, even when compiling on V2.1. It's wrong to restrict Java 8 to below 2G when customers will be running on newer OSes that support __MAP_64. Versions earlier than V2.4 are already out of support, V2.3 EOS was Oct 2022.
https://github.com/eclipse/omr/blob/779c51b9568b18dbc6650c2ca284580e65aa4792/port/unix/omrmmap.c#L208

Reply:
Is this change what you mean ?

Reply:
It's fine for shared classes, but needs to behave correctly for OMR. As discussed on Slack, it depends on the behaviour of z/OS mmap when the flag is passed in. Since we don't have all the OSes 1.13 - 2.3 to test on, the OMR code should check for < 2.4 and return an error if OMRPORT_MMAP_FLAG_ZOS_64BIT is specified.